Sources/Parameters.swift

```diff
@@ -6,16 +6,34 @@
 
 import Foundation
 
-public enum ParametersError: Error {
-  case invalidSyntax(value: Any)
-  case invalidKey(key: String, value: Any)
-  case invalidStructure(key: String, oldValue: Any, newValue: Any)
-}
+// For retro-compatibility. Remove in next major.
+@available(*, deprecated, renamed: "Parameters.Error", message: "Use the Parameters.Error nested type instead")
+public typealias ParametersError = Parameters.Error
 
+/// Namespace to handle extra context parameters passed as a list of `foo=bar` strings.
+/// Typically used when parsing command-line arguments one by one
+/// (like `foo=bar pt.x=1 pt.y=2 values=1 values=2 values=3 flag`)
+/// to turn them into a dictionary structure
 public enum Parameters {
+  public enum Error: Swift.Error {
+    case invalidSyntax(value: Any)
+    case invalidKey(key: String, value: Any)
+    case invalidStructure(key: String, oldValue: Any, newValue: Any)
+  }
+
   typealias Parameter = (key: String, value: Any)
   public typealias StringDict = [String: Any]
 
+  /// Transforms a list of strings representing structred-key/value pairs, like
+  /// `["pt.x=1", "pt.y=2", "values=1", "values=2", "values=3" "flag"]`
+  /// into a structured dictionary.
+  ///
+  /// - Parameter items: The list of `k=v`-style Strings, each string
```

Comment:
Shouldn't we use the same structure as in the other doc. blocks? Namely a `- Paramters:`, then newline, then a list of parameters (here `- items:`).